### PR TITLE
Add the FAQ

### PR DESCRIPTION
**Pull request by tpiperatgod**
Create the FAQ docs and add two questions:
1. How to use private image repositories in OpenFunction?
2. How to access the Knative-runtime function without introducing a new ingress controller?

Signed-off-by: laminar <fangtian@kubesphere.io>

### CODE REVIEW

File: docs/faq/README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# FAQ
+
+## Q: How to use private image repositories in OpenFunction?
+
+A: OpenFunction uses Shipwright (essentially using Tekton to integrate with Cloud Native Buildpacks) in the build phase to package the user function to the application image.
```
Comment (benjaminhuo) [inline docs/faq/README.md:5]: (essentially using Tekton to integrate with Cloud Native Buildpacks)
=>
(which utilizes Tekton to integrate with Cloud Native Buildpacks)

File: docs/faq/README.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+When using a private image repository, users often choose to access it in an insecure way, which is not yet supported by the Cloud Native Buildpacks.
```
Comment (benjaminhuo) [inline docs/faq/README.md:7]: When using a private image repository, users often choose to access it in an insecure way
=>
Users often choose to access a private image repository in an insecure way.

File: docs/faq/README.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+In order not to block such a need, we offer a workaround as follows:
```
Comment (benjaminhuo) [inline docs/faq/README.md:9]: We offer a workaround as below to get around this limitation for now:

File: docs/faq/README.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  apiVersion: v1
+  data:
+    Corefile: |
+      .:53 {
+          errors
+          health
+          ready
+          template IN A openfunction.dev {
+            match .*\.openfunction\.dev
+            answer "{{ .Name }} 60 IN A 10.233.47.71"
```
Comment (benjaminhuo) [inline docs/faq/README.md:66]: Need to add where this IP 10.233.47.71 comes from

**Review by benjaminhuo — CHANGES_REQUESTED** [review]

File: docs/faq/README.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+          forward . /etc/resolv.conf
+          cache 30
+          loop
+          reload
+          loadbalance
+      }
+      ...
+  ```
+
+After the above configuration is done, you can get the URL address of the function with the following command. Then you can trigger the function via `curl` or your browser.
```
Comment (benjaminhuo) [inline docs/faq/README.md:100]: Can do this without the host config?

Comment (tpiperatgod) [reply docs/faq/README.md:108]: Added instructions for this situation...

**Review by benjaminhuo — APPROVED** [review]

**Pull request merged**